tests/hf_hydrodata/test_point.py

```diff
@@ -1805,5 +1779,47 @@ def test_fail_fan_outside_timerange():
     assert str(exc.value) == "There are zero sites that satisfy the given parameters."
 
 
+def test_get_metadata_by_huc():
+    """Test for get_point_metadata with HUC filter."""
+    df = point.get_point_metadata(
+        dataset="usgs_nwis",
+        variable="streamflow",
+        temporal_resolution="daily",
+        aggregation="mean",
+        date_start="2002-01-01",
+        date_end="2002-01-05",
+        huc_id=["02040106"],
+        grid="conus2",
+    )
+    assert len(df) == 14
+    # site ID that is in bbox but not in huc
+    assert "01470736" not in list(df["site_id"])
+
+
+def test_get_data_by_huc():
```

Comment:
Again, maybe add a test or two to assert agains a not-happy path. To do this you'd probably need to mock data responses. IMO we should be doing that more for unit tests here anyway.